x/programs/cmd/simulator/cmd/plan.go

```diff
@@ -240,7 +240,15 @@ func runStepFunc(
 
 			return nil
 		}
-		id, _, balance, err := programExecuteFunc(ctx, log, db, params, method, maxUnits)
+		id, response, balance, err := programExecuteFunc(ctx, log, db, params, method, maxUnits)
+		if err != nil {
+			return err
+		}
+		resp.setResponse(response)
+		ok, err := validateAssertion(response[0], require)
```

Comment:
Is there any reason why is `response` an array ? I know that it's an initial design choice, but it doesn't seem used and we need to access some *potentially* out of bounds elements with this `0`, maybe that it will be a change of the fix for https://github.com/ava-labs/hypersdk/issues/595#event-12716725615 though

Reply:
>Is there any reason why is response an array? 

No idea

>but it doesn't seem used and we need to access some potentially out of bounds elements with this 0

🤷  haha... not an issue yet, let's fix it when it becomes an issue.

>but it doesn't seem used and we need to access some potentially out of bounds elements with this 0

Sure